src/OpenTelemetry.Collector.Dependencies/Implementation/SqlClientDiagnosticListener.cs

```diff
@@ -0,0 +1,126 @@
+            : base(sourceName, tracer)
+        {
+        }
+
+        public override void OnStartActivity(Activity activity, object payload)
+        {
+        }
+
+        public override void OnCustom(string name, Activity activity, object payload)
+        {
+            switch (name)
+            {
+                case SqlDataBeforeExecuteCommand:
+                case SqlMicrosoftBeforeExecuteCommand:
+                    {
+                        var command = this.commandFetcher.Fetch(payload);
+
+                        if (command == null)
+                        {
+                            CollectorEventSource.Log.NullPayload(nameof(SqlClientDiagnosticListener) + name);
+                            return;
+                        }
+
+                        var connection = this.connectionFetcher.Fetch(command);
+                        var database = this.databaseFetcher.Fetch(connection);
+
+                        this.Tracer.StartActiveSpan((string)database, SpanKind.Client, out var span);
+
+                        if (span.IsRecording)
+                        {
+                            var dataSource = this.dataSourceFetcher.Fetch(connection);
+                            var commandText = this.commandTextFetcher.Fetch(command);
+
+                            span.PutComponentAttribute("sql");
+
+                            span.PutDatabaseTypeAttribute("sql");
+                            span.PutDatabaseInstanceAttribute((string)database);
+                            span.PutDatabaseStatementAttribute((string)commandText);
+                            span.PutPeerServiceAttribute((string)dataSource);
+                        }
+                    }
+
+                    break;
+                case SqlDataAfterExecuteCommand:
+                case SqlMicrosoftAfterExecuteCommand:
+                    {
+                        var span = this.Tracer.CurrentSpan;
+
+                        if (span == null || !span.Context.IsValid)
+                        {
+                            CollectorEventSource.Log.NullOrBlankSpan(nameof(SqlClientDiagnosticListener) + name);
+                            return;
+                        }
+
+                        span.End();
+                    }
+
+                    break;
+                case SqlDataWriteCommandError:
+                case SqlMicrosoftWriteCommandError:
+                    {
+                        var span = this.Tracer.CurrentSpan;
+
+                        if (span == null || !span.Context.IsValid)
+                        {
+                            CollectorEventSource.Log.NullOrBlankSpan(nameof(SqlClientDiagnosticListener) + name);
+                            return;
+                        }
+
+                        if (span.IsRecording)
+                        {
+                            if (!(this.exceptionFetcher.Fetch(payload) is Exception exception))
```

Comment:
I think this would be easier to understand if an if/else was used instead of inverting the `is` check.

```csharp
if (this.exceptionFetch.Fetch(payload) is Exception exception)
{
    span.Status = Status.Unknown.WithDescription(exception.Message);
}
else
{
	CollectorEventSource.Log.NullPayload(nameof(SqlClientDiagnosticListener) + name);
}